bftengine/src/bcstatetransfer/BCStateTran.cpp

```diff
@@ -2511,6 +2512,7 @@ void BCStateTran::processData(bool lastInBatch) {
       }
     }
 
+    bool newSourceReplica = replaceSourceType != SourceReplaceType::DONT_REPLACE ? true : false;
```

Comment:
nit:
`bool newSourceReplica = (replaceSourceType != SourceReplaceType::DONT_REPLACE);`

or you can completely remove this verialbe and use replaceSourceType for the rest of the code.